### PR TITLE
Raycast fixes

### PR DESCRIPTION
**Pull request by chrisforbes**
An assortment of fixes to the raycaster and tools.

### CODE REVIEW

File: src/ship_space.cc [hunk 1]
```diff
@@ -681,7 +684,7 @@ ship_space::raycast(float ox, float oy, float oz, float dx, float dy, float dz,
             continue;
         }
 
-        if (bl->type != (block_empty ^ rc->inside)) {
+        if (rc->inside ^ (bl && bl->type != block_empty)) {
```
Comment (mkfifo) [inline src/ship_space.cc]: I would love a comment explaining this

**Comment by mkfifo** [comment]
Looking good visually, I can't compile and test this until I get home tonight.

**Comment by mkfifo** [comment]
I can both place and remove the surfaces while I am inside a cube.

Interestingly it seems I cannot interact with another cube while I am inside one cube.

**Comment by chrisforbes** [comment]
Indeed you cannot. You can only interact with the complement of the block the camera is in.

**Pull request merged**